src/main/kotlin/cacophony/controlflow/functions/GenerateCall.kt

```diff
@@ -72,14 +74,23 @@ fun generateCall(
     }
 
     nodes.add(CFGNode.Call(function))
-    nodes.add(CFGNode.AdditionAssignment(rsp, CFGNode.ConstantLazy { alignmentShift.value + stackShift }))
+    if (stackResults.isEmpty()) {
```

Comment:
I think we don't need that much ifology, we can just do `Addition(rsp, 8*stackArguments.size)->read results->Addition(rsp, alignmentShift)` - logic should remain the same